overlord/hookstate/hookmgr.go

```diff
@@ -307,10 +339,18 @@ func runHookAndWait(snapName string, revision snap.Revision, hookName, hookConte
 	// Hook completed; it may or may not have been successful.
 	case <-hookCompleted:
 		return buffer.Bytes(), hookError
-
-	// Hook was aborted.
+	// max timeout reached
+	case <-killTimerCh:
+		if err := killemAll(command); err != nil {
+			return nil, fmt.Errorf("cannot abort hook %q: %s", hookName, err)
+		}
+		return buffer.Bytes(), fmt.Errorf("exceeded maximum runtime of %s", maxRuntime)
+		// Hook was aborted.
 	case <-tomb.Dying():
-		if err := command.Process.Kill(); err != nil {
+		if killTimer != nil {
+			killTimer.Stop()
```

Comment:
there's no need for this now, we have the defer and if we got here we won't obviously execute ther ther cases of the select